dev/devicelab/lib/framework/utils.dart

```diff
@@ -0,0 +1,403 @@
+  // git show -s --format=%at 4b546df7f0b3858aaaa56c4079e5be1ba91fbb65
+  return inDirectory(flutterDirectory, () async {
+    String unixTimestamp = await eval('git', <String>[
+      'show',
+      '-s',
+      '--format=%at',
+      commit,
+    ]);
+    int secondsSinceEpoch = int.parse(unixTimestamp);
+    return new DateTime.fromMillisecondsSinceEpoch(secondsSinceEpoch * 1000);
+  });
+}
+
+Future<Process> startProcess(String executable, List<String> arguments,
+    {Map<String, String> env}) async {
+  String command = '$executable ${arguments?.join(" ") ?? ""}';
+  print('Executing: $command');
+  Process proc = await Process.start(executable, arguments,
+      environment: env, workingDirectory: cwd);
+  ProcessInfo procInfo = new ProcessInfo(command, proc);
+  _runningProcesses.add(procInfo);
+
+  proc.exitCode.then((_) {
+    _runningProcesses.remove(procInfo);
+  });
+
+  return proc;
+}
+
+Future<Null> forceQuitRunningProcesses() async {
+  // Give normally quitting processes a chance to report their exit code.
+  await new Future<Null>.delayed(const Duration(seconds: 1));
+
+  // Whatever's left, kill it.
+  for (ProcessInfo p in _runningProcesses) {
+    print('Force quitting process:\n$p');
+    if (!p.process.kill()) {
+      print('Failed to force quit process');
+    }
+  }
+  _runningProcesses.clear();
+}
+
+/// Executes a command and returns its exit code.
+Future<int> exec(String executable, List<String> arguments,
+    {Map<String, String> env, bool canFail: false}) async {
+  Process proc = await startProcess(executable, arguments, env: env);
+
+  proc.stdout
+      .transform(UTF8.decoder)
+      .transform(const LineSplitter())
+      .listen(print);
+  proc.stderr
+      .transform(UTF8.decoder)
+      .transform(const LineSplitter())
+      .listen(stderr.writeln);
+
+  int exitCode = await proc.exitCode;
+
+  if (exitCode != 0 && !canFail)
+    fail('Executable failed with exit code $exitCode.');
+
+  return exitCode;
+}
+
+/// Executes a command and returns its standard output as a String.
+///
+/// Standard error is redirected to the current process' standard error stream.
+Future<String> eval(String executable, List<String> arguments,
+    {Map<String, String> env, bool canFail: false}) async {
+  Process proc = await startProcess(executable, arguments, env: env);
+  proc.stderr.listen((List<int> data) {
+    stderr.add(data);
+  });
+  String output = await UTF8.decodeStream(proc.stdout);
+  int exitCode = await proc.exitCode;
+
+  if (exitCode != 0 && !canFail)
+    fail('Executable failed with exit code $exitCode.');
+
+  return output.trimRight();
+}
+
+Future<int> flutter(String command,
+    {List<String> options: const <String>[], bool canFail: false}) {
+  List<String> args = <String>[command]..addAll(options);
+  return exec(path.join(flutterDirectory.path, 'bin', 'flutter'), args,
+      canFail: canFail);
+}
+
+String get dartBin =>
+    path.join(flutterDirectory.path, 'bin/cache/dart-sdk/bin/dart');
```

Comment:
path.join(flutterDirectory.path, 'bin', 'cache', ...)

Not a big deal as I expect this only runs on mac/linux today, but it's consistent with above.